CMakeLists.txt

```diff
@@ -419,10 +421,19 @@ if (NOT EMSCRIPTEN)
       INSTALL
     )
 
+    if(WITH_WASI)
+      add_subdirectory("third_party/uvwasi" EXCLUDE_FROM_ALL)
+      include_directories("${libuv_SOURCE_DIR}/include")
+      include_directories(third_party/uvwasi/include)
+      add_definitions(-DWITH_WASI)
+      set(INTERP_LIBS uvwasi_a)
+    endif()
+
     # wasm-interp
     wabt_executable(
       NAME wasm-interp
-      SOURCES src/tools/wasm-interp.cc
+      SOURCES src/tools/wasm-interp.cc src/interp/interp-wasi.cc
```

Comment:
Why not make this an optional source rather than a source that is always included w/ #ifdef around it?

Reply:
Done.